### DISCUSSION TITLE
書き込んだキーマップが飛んでしまう

### DISCUSSION THREAD

**Discussion by OTR7190** [Yowkees/keyball#535] [Q&A] [answered]
Remapにてキーマップ設定して利用しておりますが利用中に設定が突然飛んでしまう事象が発生しております。
都度、再設定して利用継続しておりますがはんだ付け等の問題でしょうか。
何か原因が分かれば対処したいので、アドバイス頂けると幸いです。

**Comment by koron** [comment, marked as answer]
2つ考えられます。

1. `EE_CLR` を間違えてタイプしている
2. Pro Microの電源が低くなっている

`EE_CLR` はKeyballの設定を保存しているEEPROMを消去しリセットするキーコードです。
[Keyball61以外では via キーマップへ設定していない](https://github.com/search?q=repo%3AYowkees%2Fkeyball+EE_CLR&type=code)はずですが、
それを誤操作してしまっている場合が考えられます。

市販のPro Micro互換機には電源電圧が低くなっているものが存在します。
(参考: [type-C + 3.3V駆動の互換機の存在 / Pro Microとそのバリエーション](https://zenn.dev/koron/articles/9fee38469a8acc#type-c-%2B-3.3v%E9%A7%86%E5%8B%95%E3%81%AE%E4%BA%92%E6%8F%9B%E6%A9%9F%E3%81%AE%E5%AD%98%E5%9C%A8))
その低い電源電圧でEEPROMを操作すると設定リセットを含む
予想外の不具合を生じる可能性があります。
これに加えて、バスパワーのハブに接続し、LEDを点灯させることでも電源電圧は低下します。

なので一度、利用時の電源電圧を計測してみることをお勧めします。
それで低ければその原因に応じていくつかの対策が考えられます。

* ジャンパ(SJ1とかJ1とか書かれている)をハンダでショートさせる
* Keyballをパソコン本体に直接か、セルフパワーのハブを通じて接続する
* LEDを消灯する
